aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -637,11 +637,15 @@ void THTensor_(arange)(THTensor *r_, accreal xmin, accreal xmax, accreal step) {
   scalar_t i = 0;
 
   THArgCheck(step > 0 || step < 0, 3, "step must be nonzero");
+  THArgCheck(std::isfinite((double)xmin) && std::isfinite((double)xmax)
```

Comment:
nit: prefer C++ cast (`static_cast<double>`) over C-style cast (`(double)`)

Reply:
Also, it's definitely a bit tricky that you're converting to double here, seems worth a comment.